frontend/appflowy_flutter/lib/plugins/database_view/widgets/row/cells/url_cell/cell_editor.dart

```diff
@@ -73,7 +74,16 @@ class _URLCellEditorState extends State<URLCellEditor> {
     if (mounted) {
       if (_cellBloc.isClosed == false &&
           _controller.text != _cellBloc.state.content) {
-        _cellBloc.add(URLCellEditorEvent.updateText(_controller.text));
+        bool validURL = Uri.parse(_controller.text).isAbsolute;
```

Comment:
Are you sure you want to do `isAbsolute`? I'm not sure it does what you expect.

A link is valid with a fragment, and `isAbsolute` checks the validity of a link without a fragment. Meaning if you have a fragment in a link like this: `https://appflowy.gitbook.io/docs/essential-documentation/roadmap#guide-to-the-roadmap` where the fragment is this `#guide-to-the-roadmap` it will return false.

If the Uri is not valid, `Uri.parse` will throw a FormatException, see https://api.flutter.dev/flutter/dart-core/Uri/parse.html - That should be enough.

If you want you could extract this url check to a util class, as a method, or just as a util function loosely, like this:
```
/// Returns the parsed [Uri] if the URL is valid otherwise it
/// handles the [FormatException) and returns null 
Uri? parseValidUrl(String url) {
  try {
    return Uri.parse(url)
  } on FormatException {
    return null;
  }
}
```
Not the best tbf, normally the above code snippet would be improved by using Either from functional programming. But it's not really necessary to overcomplicate..